openlibrary/plugins/worksearch/code.py

```diff
@@ -496,16 +494,118 @@ def run_solr_query(
             params.append(('facet.field', facet))
 
     if q_list:
-        if use_dismax:
-            params.append(('q', ' '.join(q_list)))
-            params.append(('defType', 'dismax'))
-            if False:  # get_solr_next():
-                params.append(('qf', 'text alternative_title^20 author_name^20'))
+        if web.input(editions='').get('editions') == 'true':
+            EDITION_FIELDS = {
+                # Internals
+                'edition_key': 'key',
+                'text': 'text',
+                # Display data
+                'title': 'title',
+                'title_suggest': 'title_suggest',
+                'subtitle': 'subtitle',
+                'alternative_title': 'title',
+                'alternative_subtitle': 'subtitle',
+                'cover_i': 'cover_i',
+                # Misc useful data
+                'language': 'language',
+                'publisher': 'publisher',
+                'publish_date': 'publish_date',
+                'publish_year': 'publish_year',
+                # Identifiers
+                'isbn': 'isbn',
+                'id_*': 'id_*',
+                # IA
+                'ia': 'ia',
+                'ia_collection': 'ia_collection',
+                'ia_box_id': 'ia_box_id',
+            }
+            if True or use_dismax:
```

Comment:
Is this a debug if? If not,change it to just if use_dismax

Reply:
Oh yep, this is work in progress :) I _probably_ want to delete this `if` clause entirely, but not sure yet!